src-tauri/src/node.rs

```diff
@@ -261,18 +266,15 @@ fn database_config(base_path: &Path, cache_size: usize, role: &Role) -> Database
     }
 }
 
-pub async fn node_controller(path: String, node_name: String, command: &str) {
+pub(crate) async fn node_controller(path: String, node_name: String, command: NodeCommands) {
     static mut NODE_HANDLE: Option<JoinHandle<()>> = None;
     match command {
-        "start" => unsafe {
+        NodeCommands::Start => unsafe {
             NODE_HANDLE = Some(init_node(path.into(), node_name).await.unwrap());
         },
-        "restart" => unsafe {
-            drop(NODE_HANDLE.take());
+        NodeCommands::Restart => unsafe {
+            NODE_HANDLE.take().unwrap().abort();
```

Comment:
This will panic if frontend sends two `Restart` commands quickly in a row. You need `if let Some() = NODE_HANDLE.take()` here.

Reply:
That possibility slipped from my attention, thanks!